receiver/jaegerreceiver/trace_receiver.go

```diff
@@ -227,9 +233,14 @@ func (jr *jReceiver) stopTraceReceptionLocked() error {
 	jr.stopOnce.Do(func() {
 		var errs []error
 
-		if jr.agent != nil {
-			jr.agent.Stop()
-			jr.agent = nil
+		if jr.agentServer != nil {
+			if aerr := jr.agentServer.Close(); aerr != nil {
+				errs = append(errs, aerr)
+			}
+			jr.agentServer = nil
+		}
+		for _, processor := range jr.agentProcessors {
+			go processor.Stop()
```

Comment:
`TraceReceiver.Stop()` is intended to be synchronous. This violates the contract and can be source of trouble (e.g. components further in the pipeline can be stopped after this but continue to unexpectedly receive data from the receiver).
(Added an issue to document this better: https://github.com/open-telemetry/opentelemetry-collector/issues/436)

Reply:
Fixed.  Good catch.